src/houghtransform.jl

```diff
@@ -86,3 +86,112 @@ function hough_transform_standard{T<:Union{Bool,Gray{Bool}}}(
     lines
 
 end
+
+"""
+```
+circle_centers, circle_radius = hough_circle_gradient(img, scale, min_dist, canny_thres, vote_thres, min_radius, max_radius)
+```
+Returns two vectors, corresponding to circle centers and radius.
+
+The circles are generated using a hough transform variant in which a non-zero point only votes for circle
+centers perpendicular to the local gradient.
+
+Parameters:
+    img          = image to detect circles in
+    scale        = relative accumulator resolution factor
+    min_dist     = minimum distance between detected circle centers
+    canny_thres  = upper threshold for canny, lower threshold=upper threshold/4
+    vote_thres   = accumulator threshold for circle detection
+    min_radius   = minimum circle radius
+    max_radius   = maximum circle radius
+"""
+
+function hough_circle_gradient{N<:Bool, T<:Number}(
+        img_edges::AbstractArray{N,2}, img_phase::AbstractArray{T,2},
+        scale::Number, min_dist::Number,
+        vote_thres::Number, radii::AbstractVector{Int})
+
+    rows,cols=size(img_edges)
+
+    non_zeros=CartesianIndex{2}[]
+    centers=CartesianIndex{2}[]
+    circle_centers=CartesianIndex{2}[]
+    circle_radius=Int[]
+    votes=zeros(Int, Int(floor(rows/scale))+1, Int(floor(cols/scale))+1)
+
+    f = CartesianIndex(map(r->first(r), indices(votes)))
+    l = CartesianIndex(map(r->last(r), indices(votes)))
+
+    function vote!(votes, x, y)
+        fx = Int(floor(x))
+        fy = Int(floor(y))
+
+        for i in fx:fx+1
+            for j in fy:fy+1
+                if checkbounds(Bool, votes, i, j)
+                    @inbounds votes[i, j] += 1
+                end
+            end
+        end
+    end
+
+    for j in indices(img_edges, 2)::AbstractUnitRange{Int}
```

Comment:
You are right. I was trying out some performance upgrade tips and forgot to delete this.  
Deleted it now.